src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    for (const auto &flag : cmd.flags()) {
+        if (std::find(params.begin(), params.end(), flag) != params.end()) {
+            fmt::print(stderr, "missing value of {}\n", flag);
+            return false;
+        }
+
+        if (std::find(flags.begin(), flags.end(), flag) == flags.end()) {
+            fmt::print(stderr, "unknown flag {}\n", flag);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+std::ostream *get_out_stream(std::string &file_name)
+{
+    std::ostream *os_ptr = nullptr;
+    if (file_name.empty()) {
+        os_ptr = &std::cout;
+    } else {
+        os_ptr = new std::ofstream(file_name);
+        if (!*os_ptr) {
+            fprintf(stderr, "ERROR: open output file %s failed\n", file_name.c_str());
+            delete os_ptr;
+        }
+    }
+    return os_ptr;
+}
+
+bool query_disk_info(
+    shell_context *sc,
+    argh::parser &cmd,
+    std::string &node_address,
+    std::map<dsn::rpc_address, dsn::error_with<query_disk_info_response>> &err_resps,
+    std::string app_name = std::string())
+{
+    std::map<dsn::rpc_address, dsn::replication::node_status::type> nodes;
+    auto error = sc->ddl_client->list_nodes(::dsn::replication::node_status::NS_INVALID, nodes);
+    if (error != dsn::ERR_OK) {
+        fmt::print(stderr, "list nodes failed, error={}\n", error.to_string());
+        return false;
+    }
+
+    std::vector<dsn::rpc_address> targets;
+    if (!node_address.empty()) {
+        for (const auto &node : nodes) {
+            if (node.first.to_std_string() == node_address) {
+                targets.emplace_back(node.first);
+            }
+        }
+        if (targets.empty()) {
+            fmt::print(stderr, "please input valid target node_address!\n");
+            return false;
+        }
+    } else {
+        for (const auto &node : nodes) {
+            targets.emplace_back(node.first);
+        }
+    }
+
+    sc->ddl_client->query_disk_info(targets, err_resps, app_name);
+    return true;
+}
+
+bool query_disk_capacity(command_executor *e, shell_context *sc, arguments args)
+{
+    // disk_capacity [-n|--node replica_server(ip:port)] [-o|--out file_name][-j|-json][-d|--detail]
+    const std::vector<std::string> &params = {"n", "node", "o", "out"};
+    const std::vector<std::string> &flags = {"j", "json", "d", "detail"};
+    argh::parser cmd(args.argc, args.argv, argh::parser::PREFER_PARAM_FOR_UNREG_OPTION);
+    if (!parse_cmd(cmd, params, flags)) {
+        return false;
+    }
+
+    bool format_to_json = cmd[{"-j", "--json"}];
+    bool query_detail_info = cmd[{"-d", "--detail"}];
+    std::string node_address = cmd({"-n", "--node"}).str();
+    std::string file_name = cmd({"-o", "--out"}).str();
+
+    std::map<dsn::rpc_address, dsn::error_with<query_disk_info_response>> err_resps;
+    if (!query_disk_info(sc, cmd, node_address, err_resps)) {
+        return false;
+    }
+
+    std::ostream *ostream_ptr = get_out_stream(file_name);
```

Comment:
Move these code before `query_disk_info`, anything may fail locally should be check before RPC.

Reply:
Yes, these code is just refactor, and some code hasn't been check